### PR TITLE
Normalize pipeline_name for Google secrets provider

### PR DESCRIPTION
**Pull request by sultaniman**
This PR addresses the issue https://github.com/dlt-hub/dlt/issues/880 by stripping whitespaces then removing punctuation characters in `pipeline_name` before calling `get_key_name` which builds the final key name

### CODE REVIEW

File: dlt/common/configuration/providers/google_secrets.py [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
+        in_string(str): input string
+
+    Returns:
+        (str): a string without punctuatio characters and whitespaces
+    """
+    whitespace = re.compile("\s+")
+    punctuation_chars = set(string.punctuation) - {"-", "_"}
+    for char in in_string:
+        if char in punctuation_chars:
+            in_string = in_string.replace(char, "")
```
Comment (burnash) [inline dlt/common/configuration/providers/google_secrets.py:28]: Most likely can be replaces with some pre-compiled regex that will replace all punctuation chars from `in_string`.

File: dlt/common/configuration/providers/google_secrets.py [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+    Note: We exclude `_` and `-` from punctuation characters
+
+    Args:
+        in_string(str): input string
+
+    Returns:
+        (str): a string without punctuatio characters and whitespaces
+    """
+    whitespace = re.compile("\s+")
+    punctuation_chars = set(string.punctuation) - {"-", "_"}
```
[the remark this answers is not in the document]
Comment (sultaniman) [reply dlt/common/configuration/providers/google_secrets.py]: Done. 🙂

File: dlt/common/configuration/providers/google_secrets.py (hunk 1 above)
Comment (sultaniman) [reply dlt/common/configuration/providers/google_secrets.py]: Yep you are right regex suits better

File: tests/conftest.py [hunk 3]
```diff
@@ -16,12 +17,15 @@
...
+TESTS_ROOT = Path(__file__).parent.absolute()
+
+
 def initial_providers() -> List[ConfigProvider]:
     # do not read the global config
     return [
         EnvironProvider(),
-        SecretsTomlProvider(project_dir="tests/.dlt", add_global_config=False),
-        ConfigTomlProvider(project_dir="tests/.dlt", add_global_config=False),
+        SecretsTomlProvider(project_dir=str(TESTS_ROOT / ".dlt"), add_global_config=False),
```
Comment (sultaniman) [inline tests/conftest.py:27]: This one just resolves absolute path to `.dlt` config

**Review by rudolfix — APPROVED** [review]
LGTM

**Pull request merged**